src/display/annotation_layer.js

```diff
@@ -1550,13 +1606,16 @@ class AnnotationLayer {
    */
   static update(parameters) {
     for (const data of parameters.annotations) {
-      const element = parameters.div.querySelector(
+      const elements = parameters.div.querySelectorAll(
         `[data-annotation-id="${data.id}"]`
       );
-      if (element) {
-        element.style.transform = `matrix(${parameters.viewport.transform.join(
-          ","
-        )})`;
+      if (elements) {
+        elements.forEach(element => {
+          const transform = `matrix(${parameters.viewport.transform.join(
+            ","
+          )})`;
```

Comment:
It looks like the `transform`-string is constant for all elements, hence please move it out out the loop.